### ISSUE TITLE
Load too long on Blender 4.2.1 and 4.3.0 font selection

### ISSUE THREAD

**Issue by mac4kent** [samytichadou/FontSelector_blender_addon#19]
MacOS 13.6.7
Only for Mac
Make sure if you have your own MacBook Pro, iMac and etc.

Would you like to fix bug without wait too long because load make bad lag on (get extension) for Blender 4.2.1 and 4.3.0
can you do upload into extension.blender.org and fix each update more easy to upload automatic on Blender 4.2.1 and up

thanks.

**Comment by samytichadou** [comment]
Hi sorry for late answer
v2.x had a lot of issues with mac.
I just released a [new version for blender 4.2.x](https://github.com/samytichadou/FontSelector_blender_addon/releases/tag/v3_0_0_a1) (extension format), can you test and post here if it is ok ?
I don't own a mac system to test :) thanks !
https://github.com/samytichadou/FontSelector_blender_addon/releases/tag/v3_0_0_a1

**Comment by mac4kent** [comment]
I look at font but "Font Select" already there on version 3.0.0 with Blender 4.3.0 alpha.
![image](https://github.com/user-attachments/assets/e3df8c13-4852-4cd8-b693-183913394bc0)

**Comment by mac4kent** [comment]
Yes I forget where find font?

**Comment by samytichadou** [comment]
3d view header or sequencer header with a text type object selected

**Comment by mac4kent** [comment]
![image](https://github.com/user-attachments/assets/df4385d1-8067-42c1-9a8a-ef2ab27d4e4e)
that right one ?

**Comment by samytichadou** [comment]
exactly so no luck on mac for now, do you happen to know where fonts are stored on mac ?
for now i'm trying to look in `/Library/Fonts` and `/System/Library/Fonts` , do you have fonts in them ?
does your console returns a fontselector error ?

**Comment by mac4kent** [comment]
![image](https://github.com/user-attachments/assets/1f22657c-507e-4c5a-be0b-3b35deaf688b)
and
![image](https://github.com/user-attachments/assets/a5c8b4ef-6079-40c9-b72e-6b8631429855)

**Comment by mac4kent** [comment]
There is use " kentdavis@Mac ~ % cd /Volumes "
There is no C:>/ only /Volumes

**Comment by samytichadou** [comment]
thanks, the console output would also be great (you have to start blender from a terminal)

**Comment by mac4kent** [comment]
That how path correct and you may ask Nutti about that select if Windows, if macintosh, and if Linux.

**Comment by mac4kent** [comment]
of course Terminal.app

**Comment by mac4kent** [comment]
[Font Select 3.0.0.txt](https://github.com/user-attachments/files/17130102/Font.Select.3.0.0.txt)
https://youtu.be/-gIYCYaRGqQ

**Comment by mac4kent** [comment]
I do use MacBook Pro M3 Pro

**Comment by samytichadou** [comment]
thanks for all the infos, i took a look at it and try to add some debug prints for the console, can you try [with this version ](https://github.com/samytichadou/FontSelector_blender_addon/releases/download/v3_0_0_a2/font_selector_3_0_0.zip) and give me the console output ?

And after having a look at your videos, i realized it wasn't clear how the new version works :
**You don't have to set the addon preference path to a font folder**, this path is used to store a **configuration file** for fontselector to work ! The addon will collect your font folders automatically with no help from you (normally :) )
Tell me if i'm not clear !

https://github.com/samytichadou/FontSelector_blender_addon/releases/download/v3_0_0_a2/font_selector_3_0_0.zip

**Comment by mac4kent** [comment]
What is console?

**Comment by samytichadou** [comment]
> What is console?

here it is : https://docs.blender.org/manual/en/latest/advanced/command_line/launch/macos.html

And additionnaly, could you test the extension with blender 4.2.x ? 4.3.x could have some incompatibilities i am not aware of...
Thanks again

**Comment by mac4kent** [comment]
You only follow where mouse move on screeen record.
I am full deaf and I don't speak voice.

**Comment by samytichadou** [comment]
> You only follow where mouse move on screeen record. I am full deaf and I don't speak voice.

yep that is what you did on the screen record, the new version should add some print statement the extension is properly registered

**Comment by mac4kent** [comment]
good.
thank you for send me use terminal.app
there different.
file: 
[Blender 4.2 and 4.3 same 1.0.0.txt](https://github.com/user-attachments/files/17139056/Blender.4.2.and.4.3.same.1.0.0.txt)

**Comment by mac4kent** [comment]
yes there is https://extensions.blender.org/ is support add-ons and theme.
You can upload to extension on blender.org

**Comment by samytichadou** [comment]
> good. thank you for send me use terminal.app there different. file: [Blender 4.2 and 4.3 same 1.0.0.txt](https://github.com/user-attachments/files/17139056/Blender.4.2.and.4.3.same.1.0.0.txt)

This is weird, there is no trace of FONTSELECTOR ...
Could you remove the extension, close blender, re open it and reinstall it and check the console output ?
sorry for the inconvenience but i'm lost here ^^

To be sure, go to **Preferences -> Get extensions -> Down arrow next to Font Selector -> Uninstall**
Then **Preferences -> Get extensions -> Down arrow in the top right corner -> Install from disk -> select the zip file (alpha2 version)**

**Comment by mac4kent** [comment]
Yes I already did uninstall and quit then open blender and install. exactly you told me.

**Comment by samytichadou** [comment]
> Yes I already did uninstall and quit then open blender and install. exactly you told me.

hmmm, could you also try to uninstall from the **addon** panel in the preferences ? Maybe there is a confusion in blender between addon (old system) and extension (new one).
Then uninstall from **extension** panel, close and reinstall ?

**Comment by mac4kent** [comment]
add-ons and extension is same. 
add-ons doesn't have support uninstall. Only Extension because of Legacy.

**Comment by mac4kent** [comment]
![image](https://github.com/user-attachments/assets/2a7292f3-3bb8-4324-b5bb-0c3849c66a14)

![image](https://github.com/user-attachments/assets/9da78f52-93fe-4856-aba2-1c6dbb9160a6)

**Comment by mac4kent** [comment]
when you put upload to Extension only support for Blender 4.2 and up

**Comment by mac4kent** [comment]
I put all checkbox turn on of all
UI see below.

![image](https://github.com/user-attachments/assets/2564d16e-5f4b-4bef-9f3c-01cf7910674f)

and 3D Viewport Font appear.

**Comment by samytichadou** [comment]
> I put all checkbox turn on of all UI see below.
> 
> ![image](https://private-user-images.githubusercontent.com/4253305/370913661-2564d16e-5f4b-4bef-9f3c-01cf7910674f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjczMDM4OTUsIm5iZiI6MTcyNzMwMzU5NSwicGF0aCI6Ii80MjUzMzA1LzM3MDkxMzY2MS0yNTY0ZDE2ZS01ZjRiLTRiZWYtOWYzYy0wMWNmNzkxMDY3NGYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkyNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MjVUMjIzMzE1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9Y2FmYTRjY2MxMjFkOGVjYWI3M2EzZTI4YjkwNmYyNjExNjc2ZDc2OWRjYmExM2QxYjI3YTkwYTMxYjZmZjQyNiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.GLbtiTbOvnBO4WiX6Os7daHmGVdcv3hSHWXQjvoyIyY)
> 
> and 3D Viewport Font appear.

Yes, the UI can be located in several places, the user can choose, but does your list of font is filled or empty ?

**Comment by mac4kent** [comment]
Oops miss show.
take screenshot again.
![image](https://github.com/user-attachments/assets/c8bb6e23-9dff-440e-a311-a08c6a043ab6)

Also don't forget to add ~/Library/Fonts/ too

![image](https://github.com/user-attachments/assets/d94a8fb7-a760-4640-914a-bc2a1300b583)

**Comment by samytichadou** [comment]
> Oops miss show. take screenshot again. ![image](https://private-user-images.githubusercontent.com/4253305/370914925-c8bb6e23-9dff-440e-a311-a08c6a043ab6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjczMDQyNTEsIm5iZiI6MTcyNzMwMzk1MSwicGF0aCI6Ii80MjUzMzA1LzM3MDkxNDkyNS1jOGJiNmUyMy05ZGZmLTQ0MGUtYTMxMS1hMDhjNmEwNDNhYjYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkyNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MjVUMjIzOTExWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NjViOTg4YjZjMmVjZTdiMTkxYTJlNjYxOTY1MTg5OTQwNjhmMTE2M2I4YjMzZDA0MTU0NDc5ZDU5ZTk3Y2RiMCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.9RAP4lsvXuUdYv7LhOHWjh0-tvSxtc8NwnSzzv8Ea50)
> 
> Also don't forget to add ~/Library/Fonts/ too
> 
> ![image](https://private-user-images.githubusercontent.com/4253305/370914428-d94a8fb7-a760-4640-914a-bc2a1300b583.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjczMDQyNTEsIm5iZiI6MTcyNzMwMzk1MSwicGF0aCI6Ii80MjUzMzA1LzM3MDkxNDQyOC1kOTRhOGZiNy1hNzYwLTQ2NDAtOTE0YS1iYzJhMTMwMGI1ODMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkyNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MjVUMjIzOTExWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MDg2N2MyMThiYmYzODM3ZTFmZDg1YTk0ODJlNGQwZDAxOWYxNTJhZDhhNjlmYjk5NGZmNWZkZDVkMDdlOGIzOSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.KB0UD9eLQFw-TOdRosLKj7slZ6EDmfF33npgV2gOcZc)

yay ! works on mac :D 
thanks for your patience !

the fonts from `~/Library/Fonts/` should have been imported too with the alpha2 version, can you confirm they are in the blender fontselector list ?

**Comment by mac4kent** [comment]
![image](https://github.com/user-attachments/assets/51f2a712-8c2b-4c53-8943-dfe4403f0d4e)

![image](https://github.com/user-attachments/assets/2705046f-9859-42f6-9c36-9ee524634f60)

**Comment by samytichadou** [comment]
> ![image](https://private-user-images.githubusercontent.com/4253305/370916640-51f2a712-8c2b-4c53-8943-dfe4403f0d4e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjczMDQ5NDksIm5iZiI6MTcyNzMwNDY0OSwicGF0aCI6Ii80MjUzMzA1LzM3MDkxNjY0MC01MWYyYTcxMi04YzJiLTRjNTMtODk0My1kZmU0NDAzZjBkNGUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkyNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MjVUMjI1MDQ5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NjAyYzVmYTlkM2U5OWI1MGNlOGM4OTZmZjMwOTVmNjcwOGVkNjU4YmQzYmIyNTAxMDZjNjE0ODM1MGY5Zjc2MCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.lutKu8Pipsq6ByLsMucmwHMtkspLvJIPFSG3JpFpt_0)
> 
> ![image](https://private-user-images.githubusercontent.com/4253305/370916691-2705046f-9859-42f6-9c36-9ee524634f60.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjczMDQ5NDksIm5iZiI6MTcyNzMwNDY0OSwicGF0aCI6Ii80MjUzMzA1LzM3MDkxNjY5MS0yNzA1MDQ2Zi05ODU5LTQyZjYtOWMzNi05ZWU1MjQ2MzRmNjAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkyNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MjVUMjI1MDQ5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZmUwYmIyMDJiNTZhYWM5NmM0MmE0OGYwMDJjNDBhZGFiNWIyMGU3ZTk5ODk5ZDg4NDU3MzAwYzM2YTczNTliMyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.AXff6zjPeBMMn4ba7Bwp82t35jdfKp4lfxWDPo0lypI)

hmmm ok 
would you care to copy paste this in blender text editor, then run it and check the output console ?

```
import os

user_path = os.path.expanduser( '~' )
path = os.path.join(user_path, r"/Library/Fonts/")

print()
print()
print()
print("TEST CONSOLE OUTPUT")
print(path)
```

**Comment by mac4kent** [comment]
Last login: Wed Sep 25 16:51:35 on ttys000
kentdavis@Mac ~ % import os

user_path = os.path.expanduser( '~' )
path = os.path.join(user_path, r"/Library/Fonts/")

print()
print()
print()
print("TEST CONSOLE OUTPUT")
print(path)
zsh: command not found: import
zsh: unknown file attribute:  
zsh: unknown username 'er_path, r/Library/Font'
zsh: number expected
kentdavis@Mac ~ %

**Comment by samytichadou** [comment]
> Last login: Wed Sep 25 16:51:35 on ttys000 kentdavis@Mac ~ % import os
> 
> user_path = os.path.expanduser( '~' ) path = os.path.join(user_path, r"/Library/Fonts/")
> 
> print() print() print() print("TEST CONSOLE OUTPUT") print(path) zsh: command not found: import zsh: unknown file attribute: zsh: unknown username 'er_path, r/Library/Font' zsh: number expected kentdavis@Mac ~ %

thanks, i think i get it, you can try this one 
https://github.com/samytichadou/FontSelector_blender_addon/releases/download/v3_0_0_a3/font_selector_3_0_0.zip

**Comment by mac4kent** [comment]
its worked
![image](https://github.com/user-attachments/assets/bfdb0793-a426-4816-b438-22177567e2c7)
![image](https://github.com/user-attachments/assets/4b2885cb-9108-428e-80a8-47e19dd12132)

**Comment by samytichadou** [comment]
Yay ! it was a syntax error in the Mac folders list, sorry for the back and forth, and thanks for testing 
Closing the issue, I'll submit to the extension platform next week I think, I have a few things to iron out first ! Have good evening/day :)

**Comment by mac4kent** [comment]
I know about "Syntax" because of Commodore 64. but Now they lot change and I giveup.
of course take your time no problem.

**Comment by mac4kent** [comment]
You can ask Nutti (Screencast Key) and Nutti can help you to put 

if windows == ... {}
else if macintosh == ... {}
else if linux == .. {}

**Comment by samytichadou** [comment]
> You can ask Nutti (Screencast Key) and Nutti can help you to put
> 
> if windows == ... {} else if macintosh == ... {} else if linux == .. {}

Thanks I ll keep the name in mind if I have these kind of errors again !

**Comment by mac4kent** [comment]
sure.

**Comment by mac4kent** [comment]
Can you add feature when I click text automatic font select become appear for example Droid on Properties (A Text) because doesn't appear select itself.
![image](https://github.com/user-attachments/assets/98e582db-b2ef-4433-b8fd-8aad8812424f)

**Comment by samytichadou** [comment]
> Can you add feature when I click text automatic font select become appear for example Droid on Properties (A Text) because doesn't appear select itself. ![image](https://private-user-images.githubusercontent.com/4253305/370923672-98e582db-b2ef-4433-b8fd-8aad8812424f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjczMDc4NzgsIm5iZiI6MTcyNzMwNzU3OCwicGF0aCI6Ii80MjUzMzA1LzM3MDkyMzY3Mi05OGU1ODJkYi1iMmVmLTQ0MzMtYjhmZC04YWFkODgxMjQyNGYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkyNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MjVUMjMzOTM4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MDVlZmNmMWRmZjU4MTdjN2E0NzE3ODQwNmRiNGExZDU0NzJmNWNlMmMxOWM4NGEwYjBlNGEzYjEwY2UwMjg4MiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.Gz4XLVhQSKsneBKsG9A0YW6FD_hRu88XnigK_Yt2lC8)

I'm not sure what you mean ? Clicking in blender font selection panel would select the font in your finder ?

**Comment by mac4kent** [comment]
I mean your Font Selection automatic force on Font remove above for example.
![image](https://github.com/user-attachments/assets/f6161bb8-4ddf-4582-98ac-8f9997ce6ab0)
become gone font above
![image](https://github.com/user-attachments/assets/01181410-efd6-45dc-9ac2-3f04fdea44ea)

**Comment by samytichadou** [comment]
> I mean your Font Selection automatic force on Font remove above for example. ![image](https://private-user-images.githubusercontent.com/4253305/370925425-f6161bb8-4ddf-4582-98ac-8f9997ce6ab0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjczMDg3MjEsIm5iZiI6MTcyNzMwODQyMSwicGF0aCI6Ii80MjUzMzA1LzM3MDkyNTQyNS1mNjE2MWJiOC00ZGRmLTQ1ODItOThhYy04Zjk5OTdjZTZhYjAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkyNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MjVUMjM1MzQxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9M2QxZDJkYmFlMmIzOThlZTVhOWU0NjUzZDZjYzhlOTVmNWFlNWRmMDY3OWJkZTFlOTdiYThjNjY0OWM0NzBiYSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.SoXRv7SpfEUjv8PUS_MmfFvBwcEtQabW-OoqEQMukdk) become gone font above ![image](https://private-user-images.githubusercontent.com/4253305/370925709-01181410-efd6-45dc-9ac2-3f04fdea44ea.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjczMDg3MjEsIm5iZiI6MTcyNzMwODQyMSwicGF0aCI6Ii80MjUzMzA1LzM3MDkyNTcwOS0wMTE4MTQxMC1lZmQ2LTQ1ZGMtOWFjMi0zZjA0ZmRlYTQ0ZWEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkyNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MjVUMjM1MzQxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NDMzMmY5YmRlZDRmOGE4NDJmOGE3NGY0NmE4ZmI3MDFmNmZlMWQxNjk0ZmY2NjgzZGM1MTFjZjUwMzZjOTc2MSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ._IV_URZTuBA_Y3Lr64CQmL5apfxaHinaOtuqoduFEVQ)

Oh yes I see, this is a bug yes, I LL take a look tomorrow thanks.
The fonts should not be removed like this. I LL keep you posted

**Comment by mac4kent** [comment]
sureeeee

**Comment by samytichadou** [comment]
https://makertube.net/w/hHXciDmTk6YYeVBKEjpDEp

What do you think about this behavior ?

Blender is hard to follow regarding font types (bold...), so i tried to find an in between :
- Font Selector shows you all fonts (regular, bold, italic, book...)
- When you click on one which have its counterpart bold italic...in the same family, blender will load it
- If you click on a bold one, it will be loaded as regular font AND as bold font. This behavior allows user to see the bold font directly in viewport.
- You can prevent this in preferences, there is a toggle to only change "regular" font slot in blender text object

This is the only behavior which makes sense to me...

**Comment by samytichadou** [comment]
Here is a new version of font selector, with font families as first class citizen. The bug you mentioned is now taken into account through a specific property to remove or not bold/ital... fonts when changing

I let you try it and give me feedback if you have some free time, cheers ! 
:)

https://github.com/samytichadou/FontSelector_blender_addon/releases/download/v3_0_0_b1/font_selector_3_0_0_b1.zip

**Comment by samytichadou** [comment]
Giving the Bold Italic BoldItalic slots are taking into account in the new behavior of the v3.0, I'm closing this issue
